src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -567,13 +569,56 @@ private function writeProperty(&$object, $property, $value)
 
             $object->$property = $value;
         } elseif (self::ACCESS_TYPE_MAGIC === $access[self::ACCESS_TYPE]) {
-            $object->{$access[self::ACCESS_NAME]}($value);
+            $this->callMethod($object, $access[self::ACCESS_NAME], $value);
         } else {
             throw new NoSuchPropertyException($access[self::ACCESS_NAME]);
         }
     }
 
     /**
+     * Call a method and convert {@see \TypeError} to {@see InvalidArgumentException}.
+     *
+     * @param object $object
+     * @param string $method
+     * @param mixed  $value
+     *
+     * @throws InvalidArgumentException
+     * @throws \Exception
+     */
+    private function callMethod($object, $method, $value) {
+        // Cannot use class_exists because the PHP 7 polyfill defines \TypeError
+        if (PHP_MAJOR_VERSION >= 7) {
+            // PHP 7
+            try {
+                $object->{$method}($value);
+            } catch (\TypeError $e) {
+                throw new InvalidArgumentException($e->getMessage(), $e->getCode(), $e);
+            }
+
+            return;
+        }
+
+        // PHP 5
+        set_error_handler(function ($errno, $errstr) {
```

Comment:
What cases it does not handle?
If found no other messages in PHP source that match used pattern:
https://github.com/php/php-src/blob/be607e724c69c92f8cda72a45a13a26e7c439aec/Zend/zend_execute.c#L644
https://github.com/php/php-src/blob/be607e724c69c92f8cda72a45a13a26e7c439aec/Zend/zend_execute.c#L648

And it seems that message is same for all supported PHP 5 versions: https://3v4l.org/VZhkg
I prefer to use more bullet proof solution than relying that something wont happen